enumerables.rb

```diff
@@ -0,0 +1,220 @@
+module Enumerable
+  # my_each
+  def my_each()
+    return to_enum(:my_each) unless block_given?
+
+    arr = to_a if self.class == Range
+    arr = self if self.class == Array
+    arr = to_a if self.class == Hash
+    Array(self).length.times do |n|
+      yield(arr[n])
+    end
+    self
+  end
+
+  # my_each_with_index
+  def my_each_with_index()
+    return to_enum(:my_each) unless block_given?
+
+    arr = to_a if self.class == Range
+    arr = self if self.class == Array
+    arr = to_a if self.class == Hash
+    Array(self).length.times do |i|
+      yield(arr[i], i)
+    end
+    self
+  end
+
+  # my_select
+  def my_select()
+    result = []
+    return to_enum unless block_given?
+
+    Array(self).my_each do |elem|
+      next unless yield(elem)
+
+      result.push(elem)
+    end
+    result
+  end
+
+  # my_all?
+  def my_all?(arg = nil)
+    # con bloque sin argumento
+    if block_given?
+      my_each do |item|
+        return false if yield(item) == true
+      end
+    elsif if arg.class == Regexp
+            my_each do |item|
+              return false if arg.match?(item) == false
+            end
+          elsif arg.class == Class
+            my_each do |item|
+              return false if item.is_a?(arg) == false
+            end
+          elsif arg.nil?
+            my_each do |item|
+              return false unless item
+            end
+          else
+            my_each do |item|
+              return false if item != arg
+            end
+          end
+    end
+    true
+  end
```

Comment:
# Failures:

#my_all? 
1. returns true if the block never returns false or nil,
2. returns false if the block returns false or nil